src/compiler/parser.ts

```diff
@@ -436,6 +436,13 @@ module ts {
         // attached to the EOF token.
         let parseErrorBeforeNextFinishedNode: boolean = false;
 
+        export const enum StatementFlags {
+            None = 0,
+            Statement = 1,
+            ModuleElement = 2,
+            StatementOrModuleElement = Statement | ModuleElement
+        }

```

Comment:
Can we eliminate StatementOrModuleElement? I feel like it is not clear when to use ModuleElement vs StatementOrModuleElement, because they represent pretty much the same thing, but are not the same numeric value.